tests/unit/amo/components/TestMoreAddonsByAuthorsCard.js

```diff
@@ -0,0 +1,456 @@
+    });
+
+    expect(root.find(AddonsCard))
+      .toHaveProp('header', `More language packs by ${fakeAuthor.username}`);
+  });
+
+  it('shows extensions in header for ADDON_TYPE_LANG with multiple authors', () => {
+    const { store } = dispatchClientMetadata();
+    const errorHandler = createStubErrorHandler();
+    store.dispatch(addonsWithAuthorsOfType({
+      addonType: ADDON_TYPE_LANG,
+      multipleAuthors: true,
+    }));
+
+    const root = render({
+      addonType: ADDON_TYPE_LANG,
+      authorNames: [fakeAuthor.username, 'secondAuthor'],
+      errorHandler,
+      store,
+    });
+
+    expect(root.find(AddonsCard))
+      .toHaveProp('header', 'More language packs by these translators');
+  });
+
+  it('shows extensions in header for ADDON_TYPE_THEME', () => {
+    const { store } = dispatchClientMetadata();
+    const errorHandler = createStubErrorHandler();
+    store.dispatch(addonsWithAuthorsOfType({ addonType: ADDON_TYPE_THEME }));
+
+    const root = render({
+      addonType: ADDON_TYPE_THEME,
+      authorNames: [fakeAuthor.username],
+      errorHandler,
+      store,
+    });
+
+    expect(root.find(AddonsCard))
+      .toHaveProp('header', `More themes by ${fakeAuthor.username}`);
+  });
+
+  it('shows extensions in header for ADDON_TYPE_THEME with multiple authors', () => {
+    const { store } = dispatchClientMetadata();
+    const errorHandler = createStubErrorHandler();
+    store.dispatch(addonsWithAuthorsOfType({
+      addonType: ADDON_TYPE_THEME,
+      multipleAuthors: true,
+    }));
+
+    const root = render({
+      addonType: ADDON_TYPE_THEME,
+      authorNames: [fakeAuthor.username, 'secondAuthor'],
+      errorHandler,
+      store,
+    });
+
+    expect(root.find(AddonsCard))
+      .toHaveProp('header', 'More themes by these artists');
+  });
+
+  it('shows add-ons in header if no specific addonType translation found', () => {
+    const { store } = dispatchClientMetadata();
+    const errorHandler = createStubErrorHandler();
+    store.dispatch(addonsWithAuthorsOfType({
+      addonType: ADDON_TYPE_OPENSEARCH,
+    }));
+
+    const root = render({
+      addonType: ADDON_TYPE_OPENSEARCH,
+      authorNames: [fakeAuthor.username],
+      errorHandler,
+      store,
+    });
+
+    expect(root.find(AddonsCard))
+      .toHaveProp('header', `More add-ons by ${fakeAuthor.username}`);
+  });
+
+  it('shows add-ons in header if no specific addonType found with multiple authors', () => {
+    const { store } = dispatchClientMetadata();
+    const errorHandler = createStubErrorHandler();
+    store.dispatch(addonsWithAuthorsOfType({
+      addonType: ADDON_TYPE_OPENSEARCH,
+      multipleAuthors: true,
+    }));
+
+    const root = render({
+      addonType: ADDON_TYPE_OPENSEARCH,
+      authorNames: [fakeAuthor.username, 'secondAuthor'],
+      errorHandler,
+      store,
+    });
+
+    expect(root.find(AddonsCard))
+      .toHaveProp('header', 'More add-ons by these developers');
+  });
+});
```

Comment:
I'm not sure that we have coverage for the logic in `mapStateToProps` for `addons`, or is that covered in the test for the `Addon` component?

Reply:
The logic in `mapStateToProps` is covered because we dispatch actions and are not using the base component directly, but instead are using the one wrapped in `connect`.

Any test using `renderAddonsWithType` dispatches loading actions and extracts the add-ons from the store using `mapStateToProps`.

That said, we could add one more that's explicitly about that as the functionality there is totally covered but all implicitly. I'll add an extra test just to be thorough and atomic.

Reply:
Wait, scratch that, `should render a card` explicitly tests it. This is fine 😄

Reply:
My concern is that `mapStateToProps` calculates the value of the `addons` prop, but, as far as I can tell, none of the tests, including `should render a card` check that the `addons` prop receives the expected array. It's true that we exercise the function, but I don't see where we validate that it's actually working as expected.

Reply:
Oh yeah, I get what you meant. I added a test for addons, good call.